sql-plugin/src/main/320/scala/com/nvidia/spark/rapids/shims/v2/TypeSigUtilFrom320.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims.v2
+
+import com.nvidia.spark.rapids.{TypeEnum, TypeSig, TypeSigUtil}
+
+import org.apache.spark.sql.types.{DataType, DayTimeIntervalType, YearMonthIntervalType}
+
+/**
+ * Add DayTimeIntervalType and YearMonthIntervalType support
+ */
+object TypeSigUtilFrom320 extends TypeSigUtil {
+
+  override def isSupported(
+      check: TypeEnum.ValueSet,
+      dataType: DataType,
+      allowDecimal: Boolean): Boolean = {
+    dataType match {
+      case _: DayTimeIntervalType => check.contains(TypeEnum.DAYTIME)
+      case _: YearMonthIntervalType => check.contains(TypeEnum.YEARMONTH)
+      case _ => false
+    }
+  }
+
+  override def getAllSupportedTypes(): TypeEnum.ValueSet = TypeEnum.values
+
+  override def reasonNotSupported(
+      check: TypeEnum.ValueSet,
+      dataType: DataType,
+      allowDecimal: Boolean,
+      notSupportedReason: Seq[String]): Seq[String] = {
+    dataType match {
+      case _: DayTimeIntervalType =>
+        if (check.contains(TypeEnum.DAYTIME)) Seq.empty else notSupportedReason
+      case _: YearMonthIntervalType =>
+        if (check.contains(TypeEnum.YEARMONTH)) Seq.empty else notSupportedReason
+      case _ => notSupportedReason
+    }
+  }
+
+  override def getCastChecksAndSigs(
```

Comment:
I think I would prefer it if we found a way to put `daytimeChecks`, `sparkDaytimeSig` `yearmonthChecks` and `sparkYearmonthSig` inside of the `CastChecks` class.  This is just for consistency with where the rest of the checks are, and how they are overridden.

Reply:
Done. I just map the dataType to TypeEnum and invoke the existing function which accepts TypeEnum can resolve this issue.